src/ll/gcd.rs

```diff
@@ -0,0 +1,81 @@
+use std::cmp::Ordering;
+
+use ll;
+use ll::limb::Limb;
+
+pub unsafe fn gcd(gp: *mut Limb, mut ap: *mut Limb, mut an: i32, mut bp: *mut Limb, mut bn: i32) -> i32 {
+    assert!(an >= bn);
+
+    let mut gc = 0;
+    while *ap == 0 && !ll::is_zero(ap, an) && *bp == 0 && !ll::is_zero(bp, bn){
+        ap = ap.offset(1);
+        bp = bp.offset(1);
```

Comment:
Increment the gp pointer here too:

```rust
gp = gp.offset(1);
```

That will fix the problem you're having with it being 0 on 32-bit (it would still be possible to trigger the same failure on 64-bit, the test cases just happen not to).